File: Apps/Codaxy.Dextop.Showcase/Demos/Grids/ClientGridWindow.js
Ext.ns('Showcase');

Ext.define('Showcase.demos.Launcher.ClientGridWindow', {
    extend: 'Showcase.demos.Launcher',
    launch: function () {
        var remoteId = Ext.id();
        var w = Ext.create('Showcase.demos.ClientGridWindow', {
            remote: {
                remoteId: remoteId
            }
        });
        w.show();
        this.session.remote.CreateDemoWindow('ClientGridWindow', remoteId, {
			failure: function() {            
                w.close();
                alert('failed');
            }
        });
    }
});

Ext.define('Showcase.demos.ClientGridWindow', {
	extend: 'Dextop.Window',	
	width: 500,
	height: 300,
	
	title: 'Client Grid Window',
	requires: 'Ext.grid.GridPanel',
	
	initComponent: function() {			
		
		var store = Ext.create('Ext.data.Store', {
			proxy: Ext.create('Dextop.data.Proxy', {
				remote: {
					remoteId: this.remote.getSubRemoteId('store')
				},
				model: 'Showcase.demos.ClientGridWindow.model.Model'
			}),
			autoLoad: true,
			model: 'Showcase.demos.ClientGridWindow.model.Model'
		});

		var columns = Dextop.createGridColumns('Showcase.demos.ClientGridWindow.columns.Model');

		var grid = new Ext.grid.GridPanel({
			border: false,
			store: store,
			columns: columns  
		});		
		
		Ext.apply(this, {	
			layout: 'fit',
			items: grid			
		});
		
		this.callParent(arguments);	
	}
});

File: Apps/Codaxy.Dextop.Showcase/Demos/Grids/ClientGridWindow.cs
using System;
using Codaxy.Dextop.Data;

namespace Codaxy.Dextop.Showcase.Demos.Grids
{
    [Demo("ClientGridWindow",
        Title = "Client Grid",
        Description = "Create a window and grid on the client side first.", 
        ClientLauncher=true,
        Path = "~/Demos/Grids"
    )]
    [LevelAdvanced]
    [TopicDextopGrid]
    [CategoryFeature]
    public class ClientGridWindow : DextopWindow
    {
        public override void InitRemotable(DextopRemote remote, DextopConfig config)
        {
            base.InitRemotable(remote, config);                      
            Remote.AddStore("model", new Crud(), remoteId: Remote.IsClientInitiated ? "store" : null);
        }

        class Crud : DextopDataProxy<Model>
        {
            public override DextopReadResult<Model> Read(DextopReadFilter filter)
            {
                return DextopReadResult.Params(
                    new Model { Id = 1, Name = "Bill", Age = 30, Height = 180 },
                    new Model { Id = 2, Name = "Bob", Age = 26, Height = 175 }
                );
            }            
        }        

        [DextopModel]
        [DextopGrid]
        class Model
        {
            [DextopModelId]
            public int Id { get; set; }
            
            [DextopGridColumn(flex=1)]
            public String Name { get; set; }
            
            [DextopGridColumn(width = 50)]            
            public int Age { get; set; }

            [DextopGridColumn(width = 50)]
            public int Height { get; set; }
        }
    }
}